### ISSUE TITLE
SplitInto panicks when running backward

### ISSUE THREAD

**Issue by Dimev** [coreylowman/dfdx#303]
```rust
let model: SplitInto<(Linear<1, 1>, Linear<1, 1>)> = Default::default();
let (left, right) = model.forward(Tensor1D::zeros().traced());
let loss = mse_loss(right, Tensor1D::zeros());
// thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', /home/user/.cargo/git/checkouts/dfdx-318e6e5ad83eea79/3fc7be4/src/gradients.rs:273:14
let gradients = loss.backwards();
```

Seems to break here, not sure what causes it yet.
Also tried with other loss functions

**Comment by Dimev** [comment]
With help from `&'static 明旭` on the discord server, seems to be caused due to left not having any gradients

**Comment by coreylowman** [comment]
Ahh yes because `left` is unused in loss computation, great find.

Right now all the backward operations only create gradients for things that are inputs to operations (so gradients aren't created for outputs). `left` is the output of something, but not the input to anything. So during backward the gradient never gets created.

I'm thinking backward should produce a `Result` where the error could contain information about unused tensors? The other option is to just silently initialize gradients of unused tensors to zeros, but it seems something like a user might want to know.

I'm in the middle of a giant rewrite atm (draft PR will be up soon) so will take some time to get to this unfortunately

**Comment by coreylowman** [comment]
Decided to just initialize to 0s automatically. rust will probably warn if something is unused anyway. Fix is pushed to my refactor branch, so won't land for a bit
